pkg/target/pagerduty/pagerduty.go

```diff
@@ -0,0 +1,178 @@
+package pagerduty
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/PagerDuty/go-pagerduty"
+	"go.uber.org/zap"
+
+	"github.com/kyverno/policy-reporter/pkg/crd/api/policyreport/v1alpha2"
+	"github.com/kyverno/policy-reporter/pkg/target"
+	"github.com/kyverno/policy-reporter/pkg/target/formatting"
+)
+
+// Options to configure the PagerDuty target
+type Options struct {
+	target.ClientOptions
+	APIToken     string
+	ServiceID    string
+	CustomFields map[string]string
+}
+
+type client struct {
+	target.BaseClient
+	client       *pagerduty.Client
+	serviceID    string
+	customFields map[string]string
+	// Track active incidents by policy+resource
+	incidents sync.Map
+}
+
+// Create a unique key for tracking incidents
+func incidentKey(result v1alpha2.PolicyReportResult) string {
+	key := result.Policy
```

Comment:
you might want to add the rule as well. One policy can have multiple rules which fail against one resource.